website/docs/typescript.md

```diff
@@ -8,12 +8,146 @@ typed as `unknown` since they may have been populated by out of date code, or
 even from other sources. This requires you to add a type guard or similar to
 ensure the `payload` conforms to what you expect. It can be convenient to
 declare the payload types up front to avoid this `unknown`, but doing so might
-be unsafe - please be sure to read the caveats below.
+be unsafe &mdash; please be sure to read the caveats below.
 
-## `GraphileWorker.Tasks`
+## Using type guards
+
+To ensure your system is as safe as possible (and guard against old jobs, or
+jobs specified outside of TypeScript's type checking) we recommend that you use
+type guards to assert that your payload is of the expected type.
+
+```ts
+interface MyPayload {
+  username: string;
+}
+
+function assertMyPayload(payload: any): asserts payload is MyPayload {
+  if (
+    typeof payload === "object" &&
+    payload &&
+    typeof payload.username === "string"
+  ) {
+    return;
+  }
+  throw new Error("Invalid payload, expected a MyPayload");
+}
+
+const task: Task = async (payload) => {
+  assertMyPayload(payload);
+  console.log(payload.username);
+};
+```
```

Comment:
I feel like this example may not be needed any more, or could be simplified?

Reply:
Yeah this is the original example, I feel like it should stay as a simple example at the top with the explanation. How can it be stripped down?